src/FontManager.cc

```diff
@@ -1,251 +1,265 @@
-#include <stdlib.h>
-#include <node.h>
-#include <uv.h>
-#include <v8.h>
-#include <nan.h>
-#include "FontDescriptor.h"
+#include <cstdlib>
+
+#include "napi.h"
 
-using namespace v8;
+#include "FontDescriptor.h"
+#include "ResultSet.h"
+#include "FontManagerImpl.h"
 
-// these functions are implemented by the platform
-long getAvailableFonts(ResultSet **);
-long findFonts(ResultSet **, FontDescriptor *);
-long findFont(FontDescriptor **, FontDescriptor *);
-long substituteFont(FontDescriptor **, char *, char *);
+using namespace Napi;
 
 // converts a ResultSet to a JavaScript array
-Local<Array> collectResults(ResultSet *results) {
-  Nan::EscapableHandleScope scope;
-  Local<Array> res = Nan::New<Array>(results->size());
+Napi::Array collectResults(Napi::Env env, ResultSet *results) {
+  Napi::EscapableHandleScope scope(env);
+  Napi::Array res = Napi::Array::New(env, results->size());
 
   int i = 0;
   for (ResultSet::iterator it = results->begin(); it != results->end(); it++) {
-    Nan::Set(res, i++, (*it)->toJSObject());
+    res.Set(i++, (*it)->toJSObject(env));
   }
 
   delete results;
-  return scope.Escape(res);
+  return scope.Escape(res).As<Napi::Array>();
```

Comment:
Is this convertion `As<Napi::Array>()` necessary when we know `res` is already a `Napi::Array`?

Reply:
yes, see other comment.